libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp

```diff
@@ -25,14 +26,15 @@ void VotePacketHandler::process(const PacketData &packet_data, const std::shared
     return;
   }
 
-  peer->markVoteAsKnown(vote_hash);
-
-  if (vote_mgr_->voteInUnverifiedMap(vote_round, vote_hash) || vote_mgr_->voteInVerifiedMap(vote)) {
+  if (!known_votes_.insert(vote_hash)) {
     LOG(log_dg_) << "Received PBFT vote " << vote_hash << " (from " << packet_data.from_node_id_.abridged()
-                 << ") already saved in queue.";
+                 << ") already processed.";
```

Comment:
by checking the cache you only know it was received(seen). It is processed if it is inside queue